src/utils/utils.ts

```diff
@@ -62,7 +64,93 @@ function standardErrorReplacer(key: string, value: any) {
   return value;
 }
 
-function outputFormatter(msg: OutputObject): string | Uint8Array {
+async function* arrayToAsyncIterable<T>(array: T[]): AsyncIterable<T> {
+  for (const item of array) {
+    yield item;
+  }
+}
+
+// Function to handle 'table' type output
+const outputTableFormatter = async (
+  rowStream: TableRow[] | AsyncIterable<TableRow>,
+  options?: TableOptions,
+): Promise<string> => {
+  let output = '';
+  const maxColumnLengths: Record<string, number> = {};
+  let rowCount = 0;
+
+  // Initialize maxColumnLengths with header lengths if headers are provided
+  if (options?.headers) {
+    for (const header of options.headers) {
+      maxColumnLengths[header] = header.length;
+    }
+  }
+
+  let iterableStream = Array.isArray(rowStream)
+    ? arrayToAsyncIterable(rowStream)
+    : rowStream;
+
+  const updateMaxColumnLengths = (row: TableRow) => {
+    for (const [key, value] of Object.entries(row)) {
+      const cellValue =
+        value === null || value === '' || value === undefined ? 'N/A' : value;
+      maxColumnLengths[key] = Math.max(
+        maxColumnLengths[key] || 0,
+        cellValue.toString().length,
+      );
+    }
+  };
+
+  // Precompute max column lengths by iterating over the rows first
+  for await (const row of iterableStream) {
+    updateMaxColumnLengths(row);
+  }
+
+  // Reset the iterableStream if it's an array so we can iterate over it again
+  if (Array.isArray(rowStream)) {
+    iterableStream = arrayToAsyncIterable(rowStream);
+  }
+
+  // If headers are provided, add them to your output first
+  if (options?.headers) {
+    const headerRow = options.headers
+      .map((header) => header.padEnd(maxColumnLengths[header]))
+      .join('\t');
+    output += headerRow + '\n';
+  }
+
+  // Function to format a single row
+  const formatRow = (row: TableRow) => {
+    let formattedRow = '';
+
+    if (options?.includeRowCount) {
+      formattedRow += `${++rowCount}\t`;
+    }
+
+    const keysToUse = options?.headers ?? Object.keys(maxColumnLengths);
+
+    for (const key of keysToUse) {
+      const cellValue = Object.prototype.hasOwnProperty.call(row, key)
+        ? row[key]
+        : 'N/A';
```

Comment:
This is very weird. Are you just trying to convert undefined values to `'N/A'`? Just use `cellValue = row[key] ?? 'N/A';`

Reply:
Ah right, I was specifically checking for no property, vs. checking for undefined and null as well. Changed it as such